src/legacy/core_plugins/console/np_ready/public/application/components/split_panel/containers/panel_container.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { Children, ReactNode, useRef, useState } from 'react';
+
+import { PanelContextProvider } from '../context';
+import { Resizer } from '../components/resizer';
+import { PanelRegistry } from '../registry';
+
+export interface Props {
+  children: ReactNode;
+  onPanelWidthChange?: (arrayOfPanelWidths: number[]) => any;
+}
+
+interface State {
+  currentPanel: number;
+  isDragging: boolean;
+  currentResizerPos: number;
+}
+
+const initialState = { currentPanel: -1, isDragging: false, currentResizerPos: -1 };
+
+const pxToPercent = (proportion: number, whole: number) => (proportion / whole) * 100;
+
+// TODO: Write tests for this
+export function PanelsContainer({ children, onPanelWidthChange }: Props) {
+  const [firstChild, ...rest] = Children.toArray(children);
```

Comment:
This component was introduced with the idea that we could use it for n-number of things we want to have resizable - I'm more thinking of a future situation where we probably want another panel next to the two editors as part of a more holistic redesign of the UX.

<img width="1627" alt="Screenshot 2019-08-16 at 17 23 08" src="https://user-images.githubusercontent.com/8155004/64326225-e17db980-cfc9-11e9-981b-569096c5151e.png">

For simplifying this contribution, though, I don't mind limiting to 2 for now :).

As for the EUI discussion it definitely sounds like a good plan! I think it's still kind of early days with this component because I needed to figure out what exactly the things are I wanted as part of it's API (like the ability to be notified of the widths as they change and perhaps also the ability to specify custom resizers between components).